elements/lisk-transactions/src/base_transaction.ts

```diff
@@ -431,6 +433,52 @@ export abstract class BaseTransaction {
 		this._id = getId(this.getBytes());
 	}
 
+	public signAll(
+		networkIdentifier: string,
+		senderPassphrase?: string,
```

Comment:
optional? its mandatory know?

Reply:
It's not. It's mandatory for registration of multi-signature accounts after that the senderPassphrase does not have the same meaning. 

Imagine that key pair `C` was used for generating the registration and members where set to `[A,B,D]` in this case senderPassphrase could be either of `[A,B,D]` but will never be `C`.

if `C` would have been included as member then it could be one of the signers but it's not anymore senderPassphrase is just another passphrase.

Reply:
but in the case of a normal transfer transaction, for signing, we need the passphrase as mandatory know? how will that work in that case?

Reply:
In case of a transaction from regular account it needs to be passed. We probably need to add validation for either `senderPassphrase` or `passphrases` to be present. @shuse2 thought?

Reply:
i think this is `or` case, so we need to validation for each case.
if keys is specified, `passphrases` has to be greater than 1, if `keys` is not provided, ignore passphrases?

Reply:
@shuse2 need to check that proposal as `passphrases` might be 1 and be valid as it could be the case of members `[A,B,C]` and it's `C` starting the transaction

Reply:
@shuse2 @ManuGowda I've updated the code and the behaviour is now based on what's passed in: if `senderPassphrase` is provided it's assumed that it's an account with only one key pair and after signing it returns, on the other hand, if `senderPassphrase` is `undefined` and `passphrases` and `keys` are provided it follows the logic to sign in key order

Reply:
so, in this case, the commander sign should accept `senderPassphrase` as a flag or we take the first passphrase as `senderPassphrase`?

Reply:
Maybe it needs to respect the `signAll` (soon to be renamed `sign`) interface and pass either one or the other